extensions/amp-analytics/0.1/events.js

```diff
@@ -1500,35 +1504,102 @@ export class VisibilityTracker extends EventTracker {
       // false missed searches.
       const selectionMethod =
         config['selectionMethod'] || visibilitySpec['selectionMethod'];
-      unlistenPromise = this.root
-        .getAmpElement(
-          context.parentElement || context,
-          selector,
-          selectionMethod
-        )
-        .then(element => {
-          return visibilityManagerPromise.then(
-            visibilityManager => {
-              return visibilityManager.listenElement(
-                element,
-                visibilitySpec,
-                this.getReadyPromise(waitForSpec, selector, element),
-                createReportReadyPromiseFunc,
-                this.onEvent_.bind(this, eventType, listener, element)
-              );
-            },
-            () => {}
+
+      if (multiSelectorVisibilityOn && Array.isArray(selector)) {
```

Comment:
Look at the code again, I still think it's worth putting 
`selector = Array.isArray(selector) ? selector : [selector]`. And then you can keep a separate flag e.g. `isMultiSelector` and then pass that flag to `getAmpElement` call. This can helps us save the `if/else` block.

Reply:
Ok, I'm fine with this, to reduce the size of the function

Reply:
@zhouyx I took it a step further and refactored all the `unlistenPromise` into `unlistenPromises` array so we can cut down on more code.

Reply:
Cool!